src/spectral/utils/spectral-validator.js

```diff
@@ -0,0 +1,75 @@
+const MessageCarrier = require('../../plugins/utils/messageCarrier');
+const config = require('../../cli-validator/utils/processConfiguration');
+const { isOpenApiv2, isOpenApiv3 } = require('@stoplight/spectral');
+// default spectral ruleset file
+const defaultSpectralRuleset =
+  __dirname + '/../rulesets/.defaultsForSpectral.yaml';
+
+const parseResults = function(results, debug) {
+  const messages = new MessageCarrier();
+
+  if (results) {
+    for (const validationResult of results) {
+      if (validationResult) {
+        const code = validationResult['code'];
+        const severity = validationResult['severity'];
+        const message = validationResult['message'];
+        const path = validationResult['path'];
+
+        if (typeof severity === 'number' && code && message && path) {
+          if (code === 'parser') {
+            // Spectral doesn't allow disabling parser rules, so don't include them
+            // in the output (for now)
+            continue;
+          }
+          // Our validator only supports warning/error level, so only include
+          // those validation results (for now)
+          if (severity === 1) {
+            //warning
+            messages.addMessage(path, message, 'warning');
+          } else if (severity === 0) {
+            //error
+            messages.addMessage(path, message, 'error');
+          }
+        } else {
+          if (debug) {
+            console.log(
+              'There was an error while parsing the spectral results: ',
+              JSON.stringify(validationResult)
+            );
+          }
+          continue;
```

Comment:
Correct me if I'm wrong but this `continue` statement looks superfluous - there's no more code after it

Reply:
Yeah I don't think that's doing anything. I'll remove it.